features/google/src/model/importers/activities-importer-html.js

```diff
@@ -12,15 +12,91 @@ class ActivityHtmlParser {
     _extractDate(text) {
         // Ignoring the timestamp for now - we don't need hourly accuracy at the
         // time of writing this.
-        const datePattern =
-            /([A-Z][a-z][a-z]) ([0-9]{1,2}), ([0-9]{4,4}), ([0-9:]{7,8} [A|P]M) ([A-Z]{3,4})/;
-        const match = text.match(datePattern);
-        const month = match?.[1];
-        const day = match?.[2];
-        const year = match?.[3];
 
-        const dateString = `${day} ${month} ${year}`;
-        return dateString ? new Date(dateString) : null;
+        // Known formats:
+        // 3 Feb 2001, 16:05:06 UTC
+        // Feb 3, 2001, 4:05:06 PM UTC
+        // 03.02.2001, 16:05:06 UTC
+
+        const datePatterns = [
+            /([A-Z][a-z][a-z]) ([0-9]{1,2}), ([0-9]{4,4}), ([0-9:]{7,8} [A|P]M) ([A-Z]{3,4})/,
+            /([0-9]{1,2}) ([A-Z][a-z][a-z]) ([0-9]{4,4}), ([0-9:]{8,8}) ([A-Z]{3,4})/,
+            /([0-9]{1,2}).([0-9]{1,2}).([0-9]{4,4}), ([0-9:]{8,8}) ([A-Z]{3,4})/,
+        ];
+
+        // Add +1 to the index because the entire matched string will be at index 0 in match.
+        const dateMaps = [
+            {
+                day: 2,
+                month: 1,
+                year: 3,
+                time: 4,
+                timezone: 5,
+            },
+            {
+                day: 1,
+                month: 2,
+                year: 3,
+                time: 4,
+                timezone: 5,
+            },
+            {
+                day: 1,
+                month: 2,
+                year: 3,
+                time: 4,
+                timezone: 5,
+            },
+        ];
+
+        const month2Name = {
+            "01": "Jan",
+            1: "Jan",
+            "02": "Feb",
+            2: "Feb",
+            "03": "Mar",
+            3: "Mar",
+            "04": "Apr",
+            0: "Apr",
+            "05": "May",
+            5: "May",
+            "06": "Jun",
+            6: "Jun",
+            "07": "Jul",
+            7: "Jul",
+            "08": "Aug",
+            8: "Aug",
+            "09": "Sep",
+            9: "Sep",
+            10: "Oct",
+            11: "Nov",
+            12: "Dec",
+        };
+
+        let date = null;
+
+        for (let index = 0; index < datePatterns.length; index++) {
+            const match = text.match(datePatterns[index]);
+            if (match && match.length > 0) {
+                const day = match[dateMaps[index]["day"]];
+                let month = match[dateMaps[index]["month"]];
+                const year = match[dateMaps[index]["year"]];
+
+                if (!isNaN(parseInt(month))) {
+                    month = month2Name[month];
+                }
+
+                const dateString = `${day} ${month} ${year}`;
+                date = new Date(dateString);
```

Comment:
for avoiding the mapping, the `Date` constructor works also with integers...
like for
`new Date(2022, 08, 19, 07, 33, 245)`
it would output:
// Fri Aug 19 2022 07:37:05 GMT-0400 (Eastern Daylight Time)

Reply:
It does not work with `Aug` for example. So the mapping would be necessary anyway.

Reply:
with August? why?
months for the Date constructor start with `00` by the way... if that slipped you

Reply:
I am not using the date contructor with numbers. I am using strings.

Reply:
And I am not using `new Date([ array of stuff ])`. So they need to be 1 indexed because I create a natural date in Long Iso format. See long date here: https://www.w3schools.com/js/js_date_formats.asp

Reply:
> I am not using the date contructor with numbers. I am using strings.

I know... thus, I proposed to use numbers for an easier mapping here